packages/contracts/src/connect-contracts.ts

```diff
@@ -0,0 +1,141 @@
+import { Signer, Contract, providers, ethers } from 'ethers'
+import { Provider } from '@ethersproject/abstract-provider'
+import { getL1ContractData, getL2ContractData } from './contract-data'
+
+export type Network = 'goerli' | 'kovan' | 'mainnet'
+interface L1Contracts {
+  addressManager: Contract
+  canonicalTransactionChain: Contract
+  executionManager: Contract
+  fraudVerifier: Contract
+  xDomainMessenger: Contract
+  ethGateway: Contract
+  multiMessageRelayer: Contract
+  safetyChecker: Contract
+  stateCommitmentChain: Contract
+  stateManagerFactory: Contract
+  stateTransitionerFactory: Contract
+  xDomainMessengerProxy: Contract
+  l1EthGatewayProxy: Contract
+  mockBondManger: Contract
+}
+
+interface L2Contracts {
+  eth: Contract
+  xDomainMessenger: Contract
+  messagePasser: Contract
+  messageSender: Contract
+  deployerWhiteList: Contract
+  ecdsaContractAccount: Contract
+  sequencerEntrypoint: Contract
+  erc1820Registry: Contract
+  addressManager: Contract
+}
+
+/**
+ * Validates user provided a singer or provider & throws error if not
+ * @param signerOrProvider
+ */
+const checkSignerType = (signerOrProvider: Signer | Provider) => {
+  if (!signerOrProvider) {
+    throw Error('signerOrProvider argument is undefined')
+  }
+  if (
+    !Provider.isProvider(signerOrProvider) &&
+    !Signer.isSigner(signerOrProvider)
+  ) {
+    throw Error('signerOrProvider argument is the wrong type')
+  }
+}
+
+/**
+ * Connects a signer/provider to layer 1 contracts on a given network
+ * @param signerOrProvider ethers signer or provider
+ * @param network string denoting network
+ * @returns l1 contracts connected to signer/provider
+ */
+export const connectL1Contracts = async (
+  signerOrProvider,
+  network?: Network
+): Promise<L1Contracts> => {
+  checkSignerType(signerOrProvider)
+
+  if (!network) {
+    console.warn(
+      'network argument not provided to connectL1Contracts. Defaulting to mainnet.'
+    )
```

Comment:
I'm not sure how I feel about this warning - its a good idea for preventing bugs but will end up in writing to stdout when the application may not be expecting a library to write to stdout. Generally ok for browser usage though

Reply:
Yeah IMO `network` should be a required parameter if we're considering having a warning like this.